src/main/java/org/opentripplanner/raptor/path/PathBuilderLeg.java

```diff
@@ -474,20 +477,29 @@ private int transitStopArrivalTimeBefore(
    *     <li>Normal case: Walk ~ boardSlack ~ transit (access can be time-shifted)</li>
    *     <li>Flex and transit: Flex ~ (transferSlack + boardSlack) ~ transit</li>
    *     <li>Flex, walk and transit: Flex ~ Walk ~ (transferSlack + boardSlack) ~ transit</li>
+   *     <li>Flex, walk and Flex: Flex ~ Walk ~ Flex (will be timeshifted in relation to the iteration departure time)</li>
    * </ol>
    * Flex access may or may not be time-shifted.
    */
-  private void timeShiftAccessTime(RaptorSlackProvider slackProvider) {
+  private void timeShiftAccessTime(RaptorSlackProvider slackProvider, int iterationDepartureTime) {
     var accessPath = asAccessLeg().streetPath;
     var nextTransitLeg = nextTransitLeg();
 
-    @SuppressWarnings("ConstantConditions")
-    int newToTime = nextTransitLeg.transitStopArrivalTimeBefore(slackProvider, hasRides());
+    int newToTime;
 
-    if (next.isTransfer()) {
-      newToTime -= next.asTransferLeg().transfer.durationInSeconds();
+    // if there is no next transit leg then it's a Flex ~ Transfer/Walk ~ Flex path
+    if (nextTransitLeg == null) {
+      newToTime = iterationDepartureTime + accessPath.durationInSeconds();
+      if (next.isTransfer()) {
+        newToTime += next.asTransferLeg().transfer.durationInSeconds();
+      }
+    } else {
+      newToTime = nextTransitLeg.transitStopArrivalTimeBefore(slackProvider, hasRides());
+      if (next.isTransfer()) {
+        newToTime -= next.asTransferLeg().transfer.durationInSeconds();
+      }
+      newToTime = accessPath.latestArrivalTime(newToTime);
     }
-    newToTime = accessPath.latestArrivalTime(newToTime);
```

Comment:
Moving this inside the `else` does not look right - this is the code doing the time-shift for the opening-hours.

Reply:
The problem with this is that if I move it out of the else, then the test passes but the real exception comes back.

My suspicion is that path comes through raptor that shouldn't and if we want to move it out of the `else` then I need to fix _that_ problem: https://github.com/opentripplanner/OpenTripPlanner/issues/4908